rootston/cursor.c

```diff
@@ -148,14 +154,45 @@ static void roots_passthrough_cursor(struct roots_cursor *cursor,
 	}
 }
 
+static void roots_grabbed_cursor(struct roots_cursor *cursor,
+		uint32_t time) {
+	struct roots_seat *seat = cursor->seat;
+	struct wlr_surface *grabbed_surface = cursor->grabbed_surface;
+	if (grabbed_surface != NULL) {
+		double sx = seat->seat->pointer_state.sx;
+		double sy = seat->seat->pointer_state.sy;
+		double dx = cursor->cursor->x - cursor->offs_x;
+		double dy = cursor->cursor->y - cursor->offs_y;
+
+		cursor->offs_x = cursor->cursor->x;
+		cursor->offs_y = cursor->cursor->y;
```

Comment:
This isn't good, right? Unfortunately there is no easy way to translate layout coordinates to surface coordinates for some arbitrary surface.

Reply:
Actually, I'm coming around to this way because translating layout coordinates to surface coordinates for an arbitrary surface is difficult, and this is how all the other grabs operate - by comparing movement to a reference point when the grab begins.